### DISCUSSION TITLE
One quick polling question > Which operating system/s are you using to run Helix?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#7345] [Polls]
I am trying to work out what the most popular OS is for Helix users. My guess is Linux > macOS > Windows... but you never know without any real data. @the-mikedavis Would you be able to share any stats you have from the binary downloads or website visits?

I am at the early stages of preparing some learning material and wanted to know if I should focus more on providing detailed instructions for one operating system compared to another  Cheers!

**Comment by SoraTenshi** [comment]
Why radio boxes?
I use all 3...

**Reply by David-Else** [reply]
I can only find radio boxes, but I have added some more for you! :) There seems to be a limit, so I can't add all permutations.

**Comment by pascalkuthe** [comment]
I don't think we can share user statistics, everything is hosted via GH pages/releases and those don't offer any statistics. It wouldn't be GDPR compliant to collect those things without informing users/allowing opt-out AFAIK and we do none of that

**Reply by David-Else** [reply]
No problem! Hopefully a load of people will volunteer the info here... people like to promote their OS of choice... go team Linux!

**Comment by kirawi** [comment]
https://tooomm.github.io/github-release-stats/?username=helix-editor&repository=helix

It wouldn't include any source builds though.

**Reply by David-Else** [reply]
Nice one! For 23:03 that looks like:

| File Type | Downloads |
|-----------|-----------|
| Windows   | 10,004    |
| Linux     | 4,665     |
| Source    | 598       |
| macOS     | 500       |

BUT most Linux users will be downloading from a package manager and most Windows users from the binary here? These results will not be accurate?

I am wondering what features are missing on each platform? I know macOS can't do piping yet...

# Linux

    helix-23.03-aarch64-linux.tar.xz (12.09 MiB)
    Last updated on 2023-03-31 — Downloaded 482 times

    helix-23.03-riscv64-linux.tar.xz (12.27 MiB)
    Last updated on 2023-03-31 — Downloaded 27 times

    helix-23.03-x86_64-linux.tar.xz (12.13 MiB)
    Last updated on 2023-03-31 — Downloaded 2,324 times

    helix-23.03-aarch64.AppImage (20.94 MiB)
    Last updated on 2023-03-31 — Downloaded 359 times

    helix-23.03-aarch64.AppImage.zsync (0.07 MiB)
    Last updated on 2023-03-31 — Downloaded 20 times

    helix-23.03-x86_64.AppImage (20.21 MiB)
    Last updated on 2023-03-31 — Downloaded 1,420 times

    helix-23.03-x86_64.AppImage.zsync (0.07 MiB)
    Last updated on 2023-03-31 — Downloaded 33 times

                                            	
# macOS

    helix-23.03-aarch64-macos.tar.xz (11.45 MiB)
    Last updated on 2023-03-31 — Downloaded 224 times

    helix-23.03-x86_64-macos.tar.xz (11.01 MiB)
    Last updated on 2023-03-31 — Downloaded 276 times

# Windows

    helix-23.03-x86_64-windows.zip (16.65 MiB)
    Last updated on 2023-03-31 — Downloaded 10,004 times

# Source

    helix-23.03-source.tar.xz (108.36 MiB)
    Last updated on 2023-03-31 — Downloaded 598 times

**Reply by icecreammatt** [reply]
What do you mean by mac users not being able to do piping? I use the pipe symbol all the time with helix to sort lists or pretty json. By piping I mean press | in helix and then type sort or jq. Is there some other piping?

**Reply by David-Else** [reply]
I was refering to https://github.com/helix-editor/helix/pull/5468 https://github.com/helix-editor/helix/issues/2111 .

**Reply by icecreammatt** [reply]
Thanks I see. `zk` looks like a nice obsidian alternative for the CLI thanks for sharing!

**Reply by Rudxain** [reply]
> BUT most Linux users will be downloading from a package manager

As a Termux user, can confirm. Yes, I use `hx` on my Android phone 🗿

[The `.deb` is built from source](https://github.com/termux/termux-packages/commit/5724791a23c3cf0726b5ca7c22b5f94d9870cc6a) then distributed to mirrors, then distributed to end users

**Comment by abrik1** [comment]
> I am trying to work out what the most popular OS is for Helix users. My guess is Linux > macOS > Windows... but you never know without any real data. @the-mikedavis Would you be able to share any stats you have from the binary downloads or website visits?
> 
> I am at the early stages of preparing some learning material and wanted to know if I should focus more on providing detailed instructions for one operating system compared to another Cheers!



> Nice one! For 23:03 that looks like:
> File Type 	Downloads
> Windows 	10,004
> Linux 	4,665
> Source 	598
> macOS 	500
> 
> BUT most Linux users will be downloading from a package manager and most Windows users from the binary here? These results will not be accurate?
> 
> I am wondering what features are missing on each platform? I know macOS can't do piping yet...
> # Linux
> 
> ```
> helix-23.03-aarch64-linux.tar.xz (12.09 MiB)
> Last updated on 2023-03-31 — Downloaded 482 times
> 
> helix-23.03-riscv64-linux.tar.xz (12.27 MiB)
> Last updated on 2023-03-31 — Downloaded 27 times
> 
> helix-23.03-x86_64-linux.tar.xz (12.13 MiB)
> Last updated on 2023-03-31 — Downloaded 2,324 times
> 
> helix-23.03-aarch64.AppImage (20.94 MiB)
> Last updated on 2023-03-31 — Downloaded 359 times
> 
> helix-23.03-aarch64.AppImage.zsync (0.07 MiB)
> Last updated on 2023-03-31 — Downloaded 20 times
> 
> helix-23.03-x86_64.AppImage (20.21 MiB)
> Last updated on 2023-03-31 — Downloaded 1,420 times
> 
> helix-23.03-x86_64.AppImage.zsync (0.07 MiB)
> Last updated on 2023-03-31 — Downloaded 33 times
> ```
> 
> # macOS
> 
> ```
> helix-23.03-aarch64-macos.tar.xz (11.45 MiB)
> Last updated on 2023-03-31 — Downloaded 224 times
> 
> helix-23.03-x86_64-macos.tar.xz (11.01 MiB)
> Last updated on 2023-03-31 — Downloaded 276 times
> ```
> 
> # Windows
> 
> ```
> helix-23.03-x86_64-windows.zip (16.65 MiB)
> Last updated on 2023-03-31 — Downloaded 10,004 times
> ```
> 
> # Source
> 
> ```
> helix-23.03-source.tar.xz (108.36 MiB)
> Last updated on 2023-03-31 — Downloaded 598 times
> ```

The number on linux may differ because `helix` is on the standard repositories on Gentoo and Arch Linux.. I downloaded from pacman

**Reply by abrik1** [reply]
NOTE: Alpine Linux and Void Linux also provide packages for helix

**Reply by peterjc** [reply]
There are conda packages for macOS and Linux thanks to conda-forge: https://anaconda.org/conda-forge/helix
